From 195b4c770631147ccbcbadb5a80580f7a35c1a5e Mon Sep 17 00:00:00 2001
From: Jason Andryuk <jandryuk@gmail.com>
Date: Fri, 20 Sep 2019 09:10:10 -0400
Subject: [PATCH 02/26] flask: Add gethvmc to create_domain_common

When creating an hvm cpu, libxc calls vcpu_hvm().  vcpu_hvm() calls
xc_domain_hvm_getcontext() (XEN_DOMCTL_gethvmcontext) twice before
updating the hvm context and calling xc_domain_hvm_getcontext()
(XEN_DOMCTL_sethvmcontext).  Allow that operation.

Signed-off-by: Jason Andryuk <jandryuk@gmail.com>
---
 tools/flask/policy/modules/xen.if | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/flask/policy/modules/xen.if b/tools/flask/policy/modules/xen.if
index 51939a4150..03cb165263 100644
--- a/tools/flask/policy/modules/xen.if
+++ b/tools/flask/policy/modules/xen.if
@@ -58,7 +58,7 @@ define(`create_domain_common', `
 	allow $1 $2:shadow enable;
 	allow $1 $2:mmu { map_read map_write adjust memorymap physmap pinpage mmuext_op updatemp stat };
 	allow $1 $2:grant setup;
-	allow $1 $2:hvm { getparam hvmctl sethvmc
+	allow $1 $2:hvm { getparam hvmctl gethvmc sethvmc
 			setparam nested altp2mhvm altp2mhvm_op dm };
 ')
 
-- 
2.21.0

